ts/src/hashkey.ts

```diff
@@ -0,0 +1,4336 @@
+        market = this.safeMarket (marketId, market);
+        const symbol = market['symbol'];
+        return this.safePosition ({
+            'symbol': symbol,
+            'id': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'contracts': this.safeNumber (position, 'position'),
+            'contractSize': undefined,
+            'side': this.safeStringLower (position, 'side'),
+            'notional': this.safeNumber (position, 'positionValue'),
+            'leverage': this.safeInteger (position, 'leverage'),
+            'unrealizedPnl': this.safeNumber (position, 'unrealizedPnL'),
+            'realizedPnl': this.safeNumber (position, 'realizedPnL'),
+            'collateral': undefined,
+            'entryPrice': this.safeNumber (position, 'avgPrice'),
+            'markPrice': undefined,
+            'liquidationPrice': this.safeNumber (position, 'liquidationPrice'),
+            'marginMode': 'cross',
+            'hedged': true,
+            'maintenanceMargin': this.safeNumber (position, 'minMargin'),
+            'maintenanceMarginPercentage': undefined,
+            'initialMargin': this.safeNumber (position, 'margin'),
+            'initialMarginPercentage': undefined,
+            'marginRatio': undefined,
+            'lastUpdateTimestamp': undefined,
+            'lastPrice': this.safeNumber (position, 'lastPrice'),
+            'stopLossPrice': undefined,
+            'takeProfitPrice': undefined,
+            'percentage': undefined,
+            'info': position,
+        });
+    }
+
+    async fetchLeverage (symbol: string, params = {}): Promise<Leverage> {
+        /**
+         * @method
+         * @name hashkey#fetchLeverage
+         * @description fetch the set leverage for a market
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/query-futures-leverage-trade
+         * @param {string} symbol unified market symbol
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} a [leverage structure]{@link https://docs.ccxt.com/#/?id=leverage-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+        };
+        const response = await this.privateGetApiV1FuturesLeverage (this.extend (request, params));
+        //
+        //     [
+        //         {
+        //             "symbolId": "ETHUSDT-PERPETUAL",
+        //             "leverage": "5",
+        //             "marginType": "CROSS"
+        //         }
+        //     ]
+        //
+        const leverage = this.safeDict (response, 0, {});
+        return this.parseLeverage (leverage, market);
+    }
+
+    parseLeverage (leverage: Dict, market: Market = undefined): Leverage {
+        let marginMode = this.safeString (leverage, 'marginType');
+        if (marginMode !== undefined) {
```

Comment:
we can use safeStringLower here instead

Reply:
Sure, sometimes forget about useful methods